src/main/java/com/deque/axe/android/AxePropCalculator.java

```diff
@@ -0,0 +1,174 @@
+package com.deque.axe.android;
+
+import com.deque.axe.android.constants.AndroidClassNames;
+import com.deque.axe.android.utils.AxeTextUtils;
+
+import java.util.HashMap;
+import java.util.Map;
+
+class AxePropCalculator {
+
+  private static final String SPACE = " ";
+  private static Map<String, String> propMap = new HashMap<>();
+  private final String text;
+  private final String contentDescription;
+  private final String labelText;
+  private final String value;
+  private final boolean isEnabled;
+  private final String className;
+  private final String hintText;
+
+  AxePropCalculator(String text,
+                    String contentDescription,
+                    String labelText,
+                    String value,
+                    boolean isEnabled,
+                    String className,
+                    String hintText) {
+
+    this.text = text;
+    this.contentDescription = contentDescription;
+    this.labelText = labelText;
+    this.value = value;
+    this.isEnabled = isEnabled;
+    this.className = className;
+    this.hintText = hintText;
+
+    AxePropInterface axePropInterface = new AxePropInterface() {};
+    propMap.put(Props.NAME.getProp(), axePropInterface.getNameProp(
+            text,
+            contentDescription,
+            labelText,
+            hintText));
+    propMap.put(Props.ROLE.getProp(),
+            axePropInterface.getRoleProp(className));
+    propMap.put(Props.VALUE.getProp(),
+            axePropInterface.getValueProp(value));
+    propMap.put(Props.STATE.getProp(), null);
+  }
+
+  Map<String, String> getCalculatedProps() {
```

Comment:
Minor: Extract the switch case logic into separate methods for readability

Reply:
I agree it is too much to put inside it but when I tried to put the logic into a separate method it still looks too much because no matter what I have to override the respective props.

Reply:
Yeah, fair, that's why this is a personal preference. At least with the methods being extracted then I can reason about one thing at a time. Meaning I can reason about the switch case, then move to a specific use case. That said, I won't fight for this one. Give this comment a thumsb up if you want a merge, or extract the methods.

Reply:
Oh, now I get what you meant by extracting, I will extract them in new commit. Thanks 👍